packages/tools/src/tools/base/ContourSegmentationBaseTool.ts

```diff
@@ -0,0 +1,355 @@
+      segmentIndex
+    );
+
+    const segmentVisible = segmentationConfig.visibility.getSegmentVisibility(
+      toolGroupId,
+      segmentationRepresentationUID,
+      segmentIndex
+    );
+
+    // Merge the configurations from different levels based on its precedence
+    const mergedConfig = Object.assign(
+      {},
+      globalConfig?.representations?.CONTOUR ?? {},
+      toolGroupConfig?.representations?.CONTOUR ?? {},
+      segmentationRepresentationConfig?.CONTOUR ?? {},
+      segmentConfig?.CONTOUR ?? {}
+    );
+
+    let lineWidth = 1;
+    let lineDash = undefined;
+    let lineOpacity = 1;
+    let fillOpacity = 0;
+
+    if (active) {
+      lineWidth = mergedConfig.outlineWidthActive ?? lineWidth;
+      lineDash = mergedConfig.outlineDashActive ?? lineDash;
+      lineOpacity = mergedConfig.outlineOpacity ?? lineOpacity;
+      fillOpacity = mergedConfig.fillAlpha ?? fillOpacity;
+    } else {
+      lineWidth = mergedConfig.outlineWidthInactive ?? lineWidth;
+      lineDash = mergedConfig.outlineDashInactive ?? lineDash;
+      lineOpacity = mergedConfig.outlineOpacityInactive ?? lineOpacity;
+      fillOpacity = mergedConfig.fillAlphaInactive ?? fillOpacity;
+    }
+
+    lineWidth = mergedConfig.renderOutline ? lineWidth : 0;
+    fillOpacity = mergedConfig.renderFill ? fillOpacity : 0;
+
+    const color = `rgba(${segmentColor[0]}, ${segmentColor[1]}, ${segmentColor[2]}, ${lineOpacity})`;
+    const fillColor = `rgb(${segmentColor[0]}, ${segmentColor[1]}, ${segmentColor[2]})`;
+
+    return {
+      color,
+      fillColor,
+      lineWidth,
+      fillOpacity,
+      lineDash,
+      textbox: {
+        color,
+      },
+      visibility: segmentationVisible && segmentVisible,
+      locked: annotationLocked,
+    };
+  }
+
+  /**
+   * Get polyline points in canvas space
+   * @param annotation - Contour annotation
+   * @returns Polyline points in canvas space
+   */
+  protected abstract getPolylinePoints(annotation: Annotation): Types.Point2[];
+
+  /**
+   * Render the contour segmentation annotation
+   */
+  protected renderAnnotationInstance(renderContext: {
+    enabledElement: Types.IEnabledElement;
+    targetId: string;
+    annotation: Annotation;
+    annotationStyle: Record<string, any>;
+    svgDrawingHelper: SVGDrawingHelper;
+  }): boolean {
+    const { annotation, annotationStyle, svgDrawingHelper } = renderContext;
+    const { annotationUID } = annotation;
+    const polylineCanvas = this.getPolylinePoints(annotation);
+    const { lineWidth, lineDash, color, fillColor, fillOpacity } =
+      annotationStyle;
+
+    drawPolylineSvg(
+      svgDrawingHelper,
+      annotationUID,
+      'lineSegments',
+      polylineCanvas,
+      {
+        color,
+        lineDash,
+        lineWidth: Math.max(0.1, lineWidth),
+        fillColor: fillColor,
+        fillOpacity,
+      }
+    );
+
+    return true;
+  }
+
+  private _registerContourSegmentationAnnotation(
```

Comment:
Nice clear method names here make it easy to understand exactly what you are doing.